File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.js
/**
 * @param {number[]} target
 * @param {number} n
 * @return {string[]}
 */
var buildArray = function(target, n) {
  const ans = []
  const build = (start, end) => {
    while (start < end) {
      ans.push("Push")
      ans.push("Pop")
      start ++
    }
    ans.push("Push")
  }
  let pre = 1
  target.forEach(n => {
    build(pre, n)
    pre = n + 1
  })
  return ans
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.go
func buildArray(target []int, n int) []string {
  cur := 1
  ans := make([]string, 0, 2 * n)
  for _, v := range target {
    for ; cur < v; cur++ {
      ans = append(ans, "Push")
      ans = append(ans, "Pop")
    }
    ans = append(ans, "Push")
    cur++
  }
  return ans
}